.github/workflows/macos-tests.yml

```
@@ -0,0 +1,56 @@
+name: Macos tests
+
+on:
+  push:
+    branches:
+    - master
+  pull_request:
+    branches:
+    - master
```

The branch should be `main`:
```suggestion
on:
  push:
    branches:
    - main
  pull_request:
    branches:
    - main
```